integration-tests/contract-security-tickets-join.spec.ts

```diff
@@ -0,0 +1,229 @@
+          IF_NONE { FAIL } {};
+          DROP;
+          UNIT;
+          NIL operation;
+          PAIR;
+        };
+      }`,
+          init: 'None',
+        });
+
+        await opJoin.confirmation();
+        expect(opJoin.hash).toBeDefined();
+        expect(opJoin.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opJoinContract = await opJoin.contract();
+        expect(await opJoinContract.storage()).toBeTruthy();
+
+        const opCaller = await Tezos.contract.originate({
+          code: `   {  parameter address;
+    storage unit;
+    code
+      {
+        CAR; # drop storage, but leave input on stack
+        CONTRACT (option (ticket string));
+        IF_NONE { FAIL } {};
+        PUSH mutez 0;
+        PUSH nat 1;
+        PUSH string "test";
+        TICKET;
+        SOME;
+        TRANSFER_TOKENS;
+        NIL operation;
+        SWAP;
+        CONS;
+        UNIT;
+        SWAP;
+        PAIR;
+      };
+    }`,
+          init: 'Unit',
+        });
+
+        await opCaller.confirmation();
+        expect(opCaller.hash).toBeDefined();
+        expect(opCaller.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opCallerContract = await opCaller.contract();
+        expect(await opCallerContract.storage()).toBeTruthy();
+
+        await Tezos.contract
+          .at(opCallerContract.address)
+          .then((contract) => {
+            return contract.methods.default(`${opJoinContract.address}`).send();
+          })
+          .then((op) => {
+            return op.confirmation().then(() => op.hash);
+          });
+      } catch (error: any) {
+        expect(error.message).toContain('michelson_v1.invalid_primitive');
+      }
+      done();
+    });
+
+    it('Verify creating ticket is not possible with two different, but similar cty', async (done) => {
+      try {
+        const opContract = await Tezos.contract.originate({
+          code: `   { parameter unit;
+      storage unit;
+      code
+        {
+          DROP; # drop input and storage
+          PUSH nat 4;
+          PUSH string "test";
+          TICKET;
+          PUSH nat 1;
+          PUSH string "tes";
+          TICKET;
+          PAIR;
+          JOIN_TICKETS;
+          IF_NONE { FAIL } {};
+          DROP;
+          UNIT;
+          NIL operation;
+          PAIR;
+        };
+      }`,
+          init: 'Unit',
+        });
+
+        await opContract.confirmation();
+        expect(opContract.hash).toBeDefined();
+        expect(opContract.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opContractContract = await opContract.contract();
+        expect(await opContractContract.storage()).toBeTruthy();
+
+        await Tezos.contract
+          .at(opContractContract.address)
+          .then((contract) => {
+            return contract.methods.default(`Unit`).send();
+          })
+          .then((op) => {
+            return op.confirmation().then(() => op.hash);
+          });
```

Comment:
I see in most of the tests, that there is a mix between `then()` and Async/Await syntaxes. 
The following code:
```
const opContractContract = await opContract.contract();
        expect(await opContractContract.storage()).toBeTruthy();

        await Tezos.contract
          .at(opContractContract.address)
          .then((contract) => {
            return contract.methods.default(`Unit`).send();
          })
          .then((op) => {
            return op.confirmation().then(() => op.hash);
          });
```
Can be simplified as follow:
```
const opContractContract = await opContract.contract();
const op = await opContractContract.methods.default(`Unit`).send();
await op.confirmation()
```

Reply:
thanks. I got the .thens  from the Taquito doc examples

I will update the cases where thens are used